### DISCUSSION TITLE
Where is the new tokio::io -> hyper::rt wrapper type?

### DISCUSSION THREAD

**Discussion by ghost** [hyperium/hyper#3263] [General]
In this commit: https://github.com/hyperium/hyper/commit/f9f65b7aa67fa3ec0267fe015945973726285bc2 it says there's a wrapper type allowing for continued use of `tokio::io::{AsyncRead, AsyncWrite}` (which Sockets in https://github.com/cloudflare/workers-rs use). I tried adding hyper-util, but the program was no longer `wasm32-unknown-unknown` compatible. I couldn't find the related wrapper type anyway.

Could this wrapper just be set behind a feature flag instead?

**Comment by seanmonstar** [comment]
The type now exists in `hyper::rt::TokioIo`. You don't need to enable all the features to get it.

**Comment by ghost** [comment]
Wonderful. Thank you.

**Comment by ghost** [comment]
@seanmonstar Spoke too soon; I see there aren't any default features set in hyper-util, but `mio` and `socket2` are still problematic in compiling for `wasm32-unknown-unknown`... there are a bunch of `sys:: ...` items unavailable. Should that stuff be set behind the `client` feature?
